x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/TransportEqlSearchAction.java

```diff
@@ -155,6 +177,89 @@ static EqlSearchResponse createResponse(Results results, AsyncExecutionId id) {
         }
     }
 
+    private static boolean requestIsAsync(EqlSearchRequest request) {
+        return request.waitForCompletionTimeout() != null && request.waitForCompletionTimeout().getMillis() >= 0;
+    }
+
+    // can the request be proxied to the remote cluster?
+    private static boolean canMinimizeRountrips(EqlSearchRequest request, Set<String> clusterAliases) {
+        // Has minimizing the round trips been (explicitly) disabled?
+        if (request.ccsMinimizeRoundtrips() == false) {
+            return false;
+        }
+        // Is this a search against a single, remote cluster?
+        if (clusterAliases.size() != 1 || clusterAliases.contains(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY)) {
+            return false;
+        }
+        // The remote async ID would not be valid on local cluster; furthermore on results fetching we would know neither if the ID is
+        // remote or not, nor which remote cluster it belongs to (TODO: could rewrite the ID to smth like [alias:ID])
```

Comment:
I'm okay with dealing with the async bit after this gets merged. Curious how async search and CCS deals with this scenario.

Reply:
["Cross-cluster search is supported only with ccs_minimize_roundtrips set to `false`."](https://www.elastic.co/guide/en/elasticsearch/reference/current/async-search.html). I.e. it won't optimise it either, it'll query the remote cluster, but store the results locally.